modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/parameter/contributor/AccountSXPParameterContributor.java

```diff
@@ -0,0 +1,550 @@
+			else if (type == ExpandoColumnConstants.LONG_ARRAY) {
+				sxpParameters.add(
+					new LongArraySXPParameter(
+						expandoSXPParameterName, true,
+						ArrayUtil.toArray(expandoValue.getLongArray())));
+			}
+			else if (type == ExpandoColumnConstants.NUMBER) {
+				sxpParameters.add(
+					new StringSXPParameter(
+						expandoSXPParameterName, true, expandoValue.getData()));
+			}
+			else if (type == ExpandoColumnConstants.NUMBER_ARRAY) {
+				sxpParameters.add(
+					new StringArraySXPParameter(
+						expandoSXPParameterName, true,
+						StringUtil.split(expandoValue.getData())));
+			}
+			else if (type == ExpandoColumnConstants.SHORT) {
+				sxpParameters.add(
+					new IntegerSXPParameter(
+						expandoSXPParameterName, true,
+						GetterUtil.getInteger(expandoValue.getShort())));
+			}
+			else if (type == ExpandoColumnConstants.SHORT_ARRAY) {
+				short[] shortArray = expandoValue.getShortArray();
+
+				Integer[] integerArray = new Integer[shortArray.length];
+
+				for (int i = 0; i < shortArray.length; i++) {
+					integerArray[i] = (int)shortArray[i];
+				}
+
+				sxpParameters.add(
+					new IntegerArraySXPParameter(
+						expandoSXPParameterName, true, integerArray));
+			}
+			else if (type == ExpandoColumnConstants.STRING) {
+				sxpParameters.add(
+					new StringSXPParameter(
+						expandoSXPParameterName, true,
+						expandoValue.getString()));
+			}
+			else if (type == ExpandoColumnConstants.STRING_ARRAY) {
+				sxpParameters.add(
+					new StringArraySXPParameter(
+						expandoSXPParameterName, true,
+						expandoValue.getStringArray()));
+			}
+			else if (type == ExpandoColumnConstants.STRING_ARRAY_LOCALIZED) {
+				sxpParameters.add(
+					new StringArraySXPParameter(
+						StringBundler.concat(
+							expandoSXPParameterName, StringPool.UNDERLINE,
+							_language.getLanguageId(searchContext.getLocale())),
+						true,
+						expandoValue.getStringArray(
+							searchContext.getLocale())));
+			}
+			else if (type == ExpandoColumnConstants.STRING_LOCALIZED) {
+				sxpParameters.add(
+					new StringSXPParameter(
+						StringBundler.concat(
+							expandoSXPParameterName, StringPool.UNDERLINE,
+							_language.getLanguageId(searchContext.getLocale())),
+						true,
+						expandoValue.getString(searchContext.getLocale())));
+			}
+		}
+	}
+
+	private void _contribute(
+			SearchContext searchContext, Set<SXPParameter> sxpParameters)
+		throws PortalException {
+
+		long userId = searchContext.getUserId();
+
+		if (userId == 0) {
+			return;
+		}
+
+		User user = _userLocalService.fetchUserById(userId);
+
+		if (user == null) {
+			return;
+		}
+
+		long accountEntryId = GetterUtil.get(searchContext.getAttribute("accountEntryId"), -1l);
+		if(accountEntryId == -1){
+			return;
+		}
+
+		AccountEntry accountEntry = _accountEntryLocalService.fetchAccountEntry(accountEntryId);
+
+		if(accountEntry == null){
+			return;
+		}
+
+		_addExpandoSXPParameters(searchContext, sxpParameters, accountEntry);
```

Comment:
What Expando information is it contributing exactly?

Reply:
Also, where is it adding the user's account IDs to the context variables? :thinking:

Reply:
What Expando information is it contributing exactly? -- all expandos tied to the particular accountId
Also, where is it adding the user's account IDs to the context variables? -- All CP Contributors - 

CPOptionFacetsPortletSharedSearchContributor
CPSearchResultsPortletSharedSearchContributor
CPSpecificationOptionFacetsPortletSharedSearchContributor

if (accountEntry != null) {
					searchContext.setAttribute(
						"accountEntryId", accountEntry.getAccountEntryId());

Note: We finally, after discussion with the commerce team, did a different customization for this feature's purpose, but having the account as a global variable is good to have.

Reply:
```
if (accountEntry != null) {
searchContext.setAttribute(
"accountEntryId", accountEntry.getAccountEntryId());
```
I'm not sure we are talking about the same thing here. How is this code helping with having the user's account IDs in the [Predefined Context Variables](https://learn.liferay.com/w/dxp/using-search/liferay-enterprise-search/search-experiences/search-blueprints/creating-and-managing-elements/predefined-element-variables-reference)? :thinking: